File: src/commands/lb_reset.py
import nextcord
from nextcord import Interaction, Embed
from nextcord.ext import commands
from src.config.env import ADMIN_ROLE_IDS
from src.database.supabase import supabase

bot = commands.Bot(command_prefix="!")

@bot.slash_command(name="lbreset", description="Admin only: Reset all leaderboards")
async def lb_reset(interaction: Interaction):
    # Check if user is admin
    if not any(role.id in ADMIN_ROLE_IDS for role in interaction.user.roles):
        await interaction.response.send_message("❌ You are not allowed to use this command.", ephemeral=True)
        return

    # Reset customers
    supabase.table("users").update({"points": 0, "total_orders": 0, "total_successful_orders": 0}).execute()

    embed = Embed(
        title="♻️ Leaderboards Reset",
        description=f"Admin: {interaction.user.mention}\nAll customer and chef stats have been reset.",
        color=0xe74c3c
    )

    await interaction.response.send_message(embed=embed)

File: src/commands/lbReset.js
import { SlashCommandBuilder } from "discord.js";
import { ENV } from "../config/env.js";
import { supabase } from "../database/supabase.js";

export const lbReset = {
  data: new SlashCommandBuilder()
    .setName("lbreset")
    .setDescription("Admin: Reset leaderboards for all users"),

  async execute(interaction) {
    const userId = interaction.user.id;

    // Admin check
    if (!ENV.ADMIN_ROLE_IDS.includes(userId)) {
      return interaction.reply({ content: "You are not allowed to use this command.", ephemeral: true });
    }

    try {
      // Reset all points to 0 in 'users' table
      const { error } = await supabase
        .from("users")
        .update({ points: 0 });

      if (error) throw error;

      await interaction.reply({ content: "✅ Leaderboards have been reset.", ephemeral: true });
    } catch (err) {
      console.error("Leaderboard reset error:", err);
      await interaction.reply({ content: "Failed to reset leaderboards.", ephemeral: true });
    }
  }
};
